src/vtquery.cpp

```diff
@@ -268,11 +272,24 @@ struct Worker : Nan::AsyncWorker {
             // query point lng/lat geometry.hpp point (used for distance calculation later on)
             mapbox::geometry::point<double> query_lnglat{data.longitude, data.latitude};
 
-            // for each tile
+            gzip::Decompressor decompressor;
+            std::string uncompressed;
+            std::vector<std::string> buffers;
+            std::vector<std::tuple<vtzero::vector_tile, std::int32_t, std::int32_t, std::int32_t>> tiles;
+            tiles.reserve(data.tiles.size());
             for (auto const& tile_ptr : data.tiles) {
                 TileObject const& tile_obj = *tile_ptr;
-                vtzero::vector_tile tile{tile_obj.data};
-
+                if (gzip::is_compressed(tile_obj.data.data(), tile_obj.data.size())) {
+                    decompressor.decompress(uncompressed, tile_obj.data.data(), tile_obj.data.size());
+                    buffers.emplace_back(std::move(uncompressed));
+                    tiles.emplace_back(vtzero::vector_tile(buffers.back()), tile_obj.z, tile_obj.x, tile_obj.y);
+                } else {
+                    tiles.emplace_back(vtzero::vector_tile(tile_obj.data), tile_obj.z, tile_obj.x, tile_obj.y);
```

Comment:
Because `tile_obj.data` is a `protozero::data_view` (and `vtzero::data_view` is the same thing because of https://github.com/mapbox/vtzero/blob/3484299f270d3f1b3f8233bfbb7f3321ab7d1c0d/include/vtzero/types.hpp#L43). So in this case `vtzero`'s `vector_tile` constructor is fancy enough to know about this type and has a custom constructor to save us the typing (this one https://github.com/mapbox/vtzero/blob/3484299f270d3f1b3f8233bfbb7f3321ab7d1c0d/include/vtzero/vector_tile.hpp#L67-L70). But `gzip-hpp` is not so smart (because we did not want to depend directly on `protozero` in gzip-hpp to keep it light).

Overall a `string_view` or `data_view` concept is so useful I think we should plan to either:

 - 🍒 break out the `data_view` implemention in protozero and start using it everywhere. Things like gzip-hpp would then have a new dep
 - 🍎 use someone else's string view like https://abseil.io/tips/1
 - 🍊 or wait until we can use C++17 everywhere and use https://en.cppreference.com/w/cpp/header/string_view


@joto has been thinking that 🍊 is ideal and that is the reason that `protozero` has https://github.com/mapbox/protozero/blob/a44efc34e5a86e93c06390aa19c89f8e115971f6/include/protozero/data_view.hpp#L29. The idea is that you can use `protozero`'s data view until you want to opt-into using the standard, if you have it.